### PR TITLE
Add Azure > Network interface details and relationship graph

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: dashboards/network/network_interface_detail.sp [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+
+
+
+query "azure_network_interface_id_input" {
+  sql = <<-EOQ
+    select
+      title as label,
+      id as value,
+      json_build_object(
+        'Name', name,
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:116]: remove name attribute

File: dashboards/network/network_interface_detail.sp [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+      where
+        id = $1;
+  EOQ
+
+  param "id" {}
+}
+
+query "azure_network_interface_public_ip_address" {
+  sql = <<-EOQ
+  with public_ip_address_id as (
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:342]: with should be shifted by 1 tab

File: dashboards/network/network_interface_detail.sp [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+        else
+          'Disabled'
+      end as value,
+      case
+        when enable_ip_forwarding
+        then
+          'ok'
+        else
+          'alert'
+      end as type
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:379]: case can be written in single line > (Following other existing queries)
`case when enable_ip_forwarding then 'Ok' else 'Alert' end as type,`

File: dashboards/network/network_interface_detail.sp [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+  sql = <<-EOQ
+    select
+      'Accelerated Networking' as label,
+      case
+        when enable_accelerated_networking
+        then
+          'Enabled'
+        else
+          'Disabled'
+      end as value,
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:399]: can be replaced with :
`case when enable_accelerated_networking then 'Enabled' else 'Disabled' end as value,`

File: dashboards/network/network_interface_detail.sp [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+      case
+        when enable_accelerated_networking
+        then
+          'ok'
+        else
+          'alert'
+      end as type
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:406]: can be replaced with:
  `case when enable_accelerated_networking then 'Ok' else 'Alert' end as type,`

File: dashboards/network/network_interface_detail.sp [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+    where
+      id = $1;
+  EOQ
+
+  param "id" {}
+}
+
+query "azure_network_interface_tags" {
+  sql = <<-EOQ
+  select
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:438]: select should be shifted by 1 tab

File: dashboards/network/network_interface_detail.sp [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+        id = $1;
+  EOQ
+
+  param "id" {}
+}
+
+query "azure_network_interface_public_ip_address" {
+  sql = <<-EOQ
+  with public_ip_address_id as (
+  select
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:343]: select should be shifted by 1 tab

File: dashboards/network/network_interface_detail.sp [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+        }
+      }
+
+    }
+
+  }
+}
+
+
+
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:109]: Remove line 108 and 109

File: dashboards/network/docs/network_interface_detail.md [hunk 9]
```diff
@@ -0,0 +1,4 @@
+
+This dashboard answers the following questions:
+
+<!-- to do -->
```
Comment (madhushreeray30) [inline dashboards/network/docs/network_interface_detail.md:4]: Add the points like :
- How is the network interface detail configured?
- What tags are applied?
- Which virtual machines, public ip and network security group is it attached to?

File: dashboards/network/network_interface_detail.sp [hunk 10]
```diff
@@ -0,0 +1,450 @@
...
+  sql = <<-EOQ
+  with vm_network_interface_id as (
+      select
+        id,
+        name,
+        subscription_id,
+        resource_group,
+        title,
+        region,
+        jsonb_array_elements(network_interfaces)->>'id' as n_id
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:193]: use `nid` instead of `n_id`

File: dashboards/network/network_interface_detail.sp [hunk 11]
```diff
@@ -0,0 +1,450 @@
...
+
+edge "azure_network_interface_from_compute_virtual_machine_edge" {
+  title = "attached to"
+
+  sql = <<-EOQ
+  with vm_network_interface_id as (
+      select
+        id,
+        name,
+        jsonb_array_elements(network_interfaces)->>'id' as n_id
```
Comment (madhushreeray30) [inline dashboards/network/network_interface_detail.sp:226]: use `nid` as `n_id`

File: dashboards/network/network_interface_detail.sp [hunk 12]
```diff
@@ -0,0 +1,450 @@
...
+
+query "azure_network_interface_id_input" {
+  sql = <<-EOQ
+    select
+      title as label,
+      id as value,
+      json_build_object(
+        'Name', name,
+        'ID', id,
+        'Subscription ID', subscription_id,
```
[the remark this answers is not in the document]
Comment (Priyanka-Chatterjee-2000) [reply dashboards/network/network_interface_detail.sp]: do you mean this-> 'subscription' , subscription_id

File: dashboards/network/network_interface_detail.sp (hunk 10 above)
Comment (Priyanka-Chatterjee-2000) [reply dashboards/network/network_interface_detail.sp:202]: deliberately kept that way as v.nid  =  n.id confuses, just  mark a difference kept it v.n_id = n.id shall i change?

**Pull request merged**